### ISSUE TITLE
import/named - False positive after upgrading from 2.23.4 to  2.24.0+ with @octokit/types

### ISSUE THREAD

**Issue by WtfJoke** [import-js/eslint-plugin-import#2445]
After upgrading from 2.23.4 to 2.24.0 onwards I experience a false positive with the rule `import/named`.

In this code snippet:
```ts
import {Endpoints} from '@octokit/types'
import {GitHub} from '@actions/github/lib/utils'

export type GithubReleaseAssets =
  Endpoints['GET /repos/{owner}/{repo}/releases/{release_id}']['response']['data']['assets']
export type GithubOktokit = InstanceType<typeof GitHub>
```
Following issue:
```
github-types.ts
  1:9  error  Endpoints not found in '@octokit/types'  import/named
```

The interface however is exported from `@octokit/types` if you take a look at: [https://github.com/octokit/types.ts/src/index.ts#L21](https://github.com/octokit/types.ts/blob/7e5dd312188253e962fa209b16963f78113ba8c3/src/index.ts#L21) and the corresponding file [https://github.com/octokit/types.ts//src/generated/Endpoints.ts#L107](https://github.com/octokit/types.ts/blob/7e5dd312188253e962fa209b16963f78113ba8c3/src/generated/Endpoints.ts#L107)

Project, where you can reproduce the error is here:
Working with [2.23.4](https://github.com/WtfJoke/setup-tectonic/tree/eslint-import-working), false positive on [2.24.0](https://github.com/WtfJoke/setup-tectonic/tree/eslint-import-false-positive)

**Comment by ljharb** [comment]
https://unpkg.com/browse/@octokit/types@6.34.0/package.json points to https://unpkg.com/browse/@octokit/types@6.34.0/dist-node/index.js which indeed does not have that entry point.

If it’s a type, use `import type`, and this plugin will ignore it.

**Comment by WtfJoke** [comment]
Thanks for your fast respone.

> https://unpkg.com/browse/@octokit/types@6.34.0/package.json points to https://unpkg.com/browse/@octokit/types@6.34.0/dist-node/index.js which indeed does not have that entry point.

Not sure how unpkg gets that info. But do you want to say its an issue in the `@oktokit/types` package and I should open an issue there?



> If it’s a type, use `import type`, and this plugin will ignore it.

Its not a type, its an interface, see https://github.com/octokit/types.ts/blob/master/src/generated/Endpoints.ts#L107

**Comment by ljharb** [comment]
unpkg is just showing what’s in the package; that’s the info node gets.

i don’t think it’s an issue in that package. An interface is a type, not a value.

**Comment by WtfJoke** [comment]
Thanks a lot! I wasnt aware of that. I use import type now
